src/textual/widgets/_option_list.py

```diff
@@ -0,0 +1,943 @@
+        if self._options[option_index].disabled:
+            # Disabled but the highlight?
+            if option_index == highlighted:
+                return strip.apply_style(
+                    self.get_component_rich_style(
+                        "option-list--option-hover-highlighted-disabled"
+                        if option_index == mouse_over
+                        else "option-list--option-highlighted-disabled"
+                    )
+                )
+            # Disabled but mouse hover?
+            if option_index == mouse_over:
+                return strip.apply_style(
+                    self.get_component_rich_style("option-list--option-hover-disabled")
+                )
+            # Just a normal disabled option.
+            return strip.apply_style(
+                self.get_component_rich_style("option-list--option-disabled")
+            )
+
+        # Handle drawing a highlighted option.
+        if highlighted is not None and line_number in spans[highlighted]:
+            # Highlighted with the mouse over it?
+            if option_index == mouse_over:
+                return strip.apply_style(
+                    self.get_component_rich_style(
+                        "option-list--option-hover-highlighted"
+                    )
+                )
+            # Just a normal highlight.
+            return strip.apply_style(
+                self.get_component_rich_style("option-list--option-highlighted")
+            )
+
+        # Perhaps the line is within an otherwise-uninteresting option that
+        # has the mouse hovering over it?
+        if mouse_over is not None and line_number in spans[mouse_over]:
+            return strip.apply_style(
+                self.get_component_rich_style("option-list--option-hover")
+            )
+
+        # It's a normal option line.
+        return strip.apply_style(self.rich_style)
+
+    def scroll_to_highlight(self) -> None:
+        """Ensure that the highlighted option is in view."""
+        highlighted = self.highlighted
+        if highlighted is None:
+            return
+        try:
+            span = self._spans[highlighted]
+        except IndexError:
+            # Index error means we're being asked to scroll to a highlight
+            # before all the tracking information has been worked out.
+            # That's fine; let's just NoP that.
+            return
+        self.scroll_to_region(
+            Region(
+                0, span.first, self.scrollable_content_region.width, span.line_count
+            ),
+            force=True,
+            animate=self.show_vertical_scrollbar,  # https://github.com/Textualize/textual/issues/2077
+        )
+
+    def validate_highlighted(self, highlighted: int | None) -> int | None:
+        """Validate the `highlighted` property value on access."""
+        if not self._options:
+            return None
+        if highlighted is None or highlighted < 0:
+            return 0
```

Comment:
I actually expected this to be
```py
if not self._options or highlighted is None:
    return None
if highlighted < 0:
    return None
```

It is a bit weird that `highlighted` _can_ be `None` but _I_ can't set it to `None` to remove highlighting.